File: 프로그래머스/1/178871. 달리기 경주/달리기 경주.java
import java.util.*;
class Solution {
    public String[] solution(String[] players, String[] callings) {
        HashMap<String,Integer> playerMap = new HashMap<>();
            for(int i=0; i<players.length; i++){
                playerMap.put(players[i],i);
            }
        
        for(String calling : callings){
            int currentIndex = playerMap.get(calling);
            if(currentIndex > 0){
                int prevIndex = currentIndex -1;
                
                String temp = players[prevIndex];
                players[prevIndex] = players[currentIndex];
                players[currentIndex] = temp;
                
                playerMap.put(calling,prevIndex);
                playerMap.put(temp,currentIndex);
            }
        }
        return players;
    }
}

File: 프로그래머스/1/178871. 달리기 경주/달리기 경주.py
def solution(players, callings):
    pos = {}
    for idx,player in enumerate(players):
        pos[player] = idx
    
    for name in callings:
        idx = pos[name]
        front = players[idx-1]
        
        players[idx] , players[idx-1] = players[idx-1] , players[idx]
        
        pos[front] = idx
        pos[name] = idx-1

    return players
        
        
        
